active/0000-keyword-arguments.md

```diff
@@ -0,0 +1,139 @@
+- Start Date: 2015-01-26
+- RFC PR #: (leave this empty)
+- Rust Issue #: (leave this empty)
+
+
+# Summary
+
+Add keyword arguments to Rust in a backwards-compatible way.
+This allows overloading that is safe in regards to type inference while being decided at compile time.
+
+# Motivation
+
+Allow another kind of argument in Rust. Current arguments distinguish themselves from other types by their order.
+If there is no semantic reason for a certain order, the argument order in functions is basically arbitrary.
+Consider a call like this:
+```rust
+window.addNewControl("Title", 20, 50, 100, 50, true);
+```
+
+First of all, the call tells nothing the user about what those values mean.
+Secondly, their order is arbitrary and must be memorized. What I propose is that this call would rather look like:
+```rust
+window.addNewControl(
+    title => "Title", 
+    xPosition => 20, 
+    yPosition => 50, 
+    width => 100, 
+    height => 50, 
+    drawingNow => true);
+```
+
+While you might argue that this is more verbose, this is already the standard in the JavaScript ecosystem.
+A lot of libraries in JavaScript actually have a convention with calling with associative arrays like:
+```JavaScript
+window.addNewControl({ xPosition: 20, yPosition: 50, width: 100, height: 5,
+                 drawingNow: true });
+```
+If this wasn't a convenient pattern, nobody would bother to do it.
```

Comment:
Because you have to declare a struct manually, while in JavaScript everything is ducktyped. It is considerably more difficult to do the same thing in Rust. Also, by allowing keywords you get overloading.